utilities/transactions/write_prepared_txn.cc

```diff
@@ -355,9 +350,16 @@ Status WritePreparedTxn::RollbackInternal() {
   // Mark the txn as rolled back
   uint64_t& rollback_seq = seq_used;
   if (s.ok()) {
+    // Note: it is safe to do it after PreReleaseCallback via WriteImpl since
+    // all the writes by the prpared batch are already blinded by the rollback
+    // batch. The only reason we commit the prepared batch here is to benefit
+    // from the existing mechanism in CommitCache that takes care of the rare
+    // cases that the prepare seq is visible to a snsapshot but max evicted seq
+    // advances that prepare seq.
     for (size_t i = 0; i < prepare_batch_cnt_; i++) {
-      wpt_db_->RollbackPrepared(GetId() + i, rollback_seq);
+      wpt_db_->AddCommitted(GetId() + i, rollback_seq);
```

Comment:
In line 317 the prepare is committed. Why we commit it again here? Do I miss anything?

Reply:
We get to here only if do_one_write is false. In that case the callback you are referring to is not passed to WriteImpl; instead a nullptr is passed at  Line 324.